cmd/catalyst-node/catalyst-node.go

```diff
@@ -0,0 +1,284 @@
+		Addr:    serfRPCAddress,
+		AuthKey: serfRPCAuthKey,
+	})
+}
+
+func getSerfMembers(client *serfclient.RPCClient) ([]serfclient.Member, error) {
+	return client.Members()
+}
+
+func changeLoadBalancerServers(endpoint string, server string, action string) ([]byte, error) {
+	url := endpoint + "?" + action + "server=" + url.QueryEscape(server)
+	req, err := http.NewRequest("POST", url, nil)
+	if err != nil {
+		glog.Errorf("Error creating request: %v", err)
+		return nil, err
+	}
+
+	client := &http.Client{}
+	resp, err := client.Do(req)
+	if err != nil {
+		glog.Errorf("Error making request: %v", err)
+		return nil, err
+	}
+
+	b, err := ioutil.ReadAll(resp.Body)
+
+	if err != nil {
+		glog.Errorf("Error reading response: %v", err)
+		return nil, err
+	}
+
+	if resp.StatusCode != http.StatusOK {
+		glog.Errorf("Error response from load balancer changing servers: %s\n", string(b))
+		return b, errors.New(string(b))
+	}
+
+	glog.V(6).Infof("requested mist to %s server %s to the load balancer\n", action, server)
+	glog.V(6).Infof(string(b))
+	return b, nil
+}
+
+func getMistLoadBalancerServers(endpoint string) (map[string]interface{}, error) {
+
+	url := endpoint + "?lstservers=1"
+	req, err := http.NewRequest("GET", url, nil)
+	if err != nil {
+		glog.Errorf("Error creating request: %v", err)
+		return nil, err
+	}
+
+	client := &http.Client{}
+	resp, err := client.Do(req)
+	if err != nil {
+		glog.Errorf("Error making request: %v", err)
+		return nil, err
+	}
+
+	if resp.StatusCode != http.StatusOK {
+		b, _ := ioutil.ReadAll(resp.Body)
+		glog.Errorf("Error response from load balancer listing servers: %s\n", string(b))
+		return nil, errors.New(string(b))
+	}
+	b, err := ioutil.ReadAll(resp.Body)
+
+	if err != nil {
+		glog.Errorf("Error reading response: %v", err)
+		return nil, err
+	}
+
+	var mistResponse map[string]interface{}
+
+	json.Unmarshal([]byte(string(b)), &mistResponse)
+
+	return mistResponse, nil
+}
+
+func main() {
+	args := os.Args[1:]
+
+	flag.Set("logtostderr", "true")
+	vFlag := flag.Lookup("v")
+	fs := flag.NewFlagSet("catalyst-node-connected", flag.ExitOnError)
+
+	verbosity := fs.String("v", "", "Log verbosity.  {4|5|6}")
+	serfRPCAddress := fs.String("serf-rpc-address", "127.0.0.1:7373", "Serf RPC address")
+	serfRPCAuthKey := fs.String("serf-rpc-auth-key", "", "Serf RPC auth key")
+	mistLoadBalancerEndpoint := fs.String("mist-load-balancer-endpoint", "http://127.0.0.1:8042/", "Mist util load endpoint")
+
+	ff.Parse(
+		fs, os.Args[1:],
+		ff.WithConfigFileFlag("config"),
+		ff.WithConfigFileParser(ff.PlainParser),
+		ff.WithEnvVarPrefix("CATALYST_NODE"),
+		ff.WithEnvVarSplit(","),
+	)
+	vFlag.Value.Set(*verbosity)
+	flag.CommandLine.Parse(nil)
+
+	config := catalystConfig{
+		serfRPCAddress:           *serfRPCAddress,
+		serfRPCAuthKey:           *serfRPCAuthKey,
+		mistLoadBalancerEndpoint: *mistLoadBalancerEndpoint,
+	}
+
+	go func() {
+		// eli note: i put this in a loop in case client boots before server.
```

Comment:
How about using a waitgroup (or `struct{}` channel) instead of this command and `time.Sleep()`?